src/main/kotlin/frc/robot/subsystems/drive/Drive.java

```diff
@@ -134,16 +140,17 @@ public class Drive extends SubsystemBase {
     private final SysIdRoutine turnSysId;
     private final Alert gyroDisconnectedAlert =
             new Alert("Disconnected gyro, using kinematics as fallback.", AlertType.kError);
-    @AutoLogOutput private Rotation2d desiredHeading;
+    @AutoLogOutput
+    private Rotation2d desiredHeading;
 
     public SwerveDriveKinematics kinematics = new SwerveDriveKinematics(getModuleTranslations());
     private Rotation2d rawGyroRotation = new Rotation2d();
     public SwerveModulePosition[] lastModulePositions = // For delta tracking
-            new SwerveModulePosition[] {
-                new SwerveModulePosition(),
-                new SwerveModulePosition(),
-                new SwerveModulePosition(),
-                new SwerveModulePosition()
+            new SwerveModulePosition[]{
```

Comment:
GitHub doesn't let me comment on these lines, so about `slewRateLimiterX` and `slewRateLimiterY`:
1. Qodana says they can be final
2. `1.5` is a magic number